### PR TITLE
[runtime-security] use remote tagger to populate event

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

Use remote tagger to populate events. one can now write rule like
```
expression: open.file.path == "/etc/passwd" && "image_name:busybox" in container.tags
```
### Motivation

What inspired you to submit this pull request?

### Additional Notes

This PR is required :
https://github.com/DataDog/datadog-agent/pull/7814

### Describe your test plan

Container tags should be part of the generated events

### CODE REVIEW

File: .gitlab/source_test/ebpf.yml [hunk 1]
```diff
@@ -35,6 +35,7 @@ tests_ebpf:
     # Compile runtime security stress tests to be executed in kitchen tests
     - inv -e security-agent.build-stress-tests --output=$DD_AGENT_TESTING_DIR/site-cookbooks/dd-security-agent-check/files/stresssuite
     # Compile master version for comparison, uncomment following lines when merged
+    - git diff
```
Comment (hmahmood) [inline .gitlab/source_test/ebpf.yml:38]: Is this required or is it there just for debugging?

File: pkg/security/probe/model.go [hunk 2]
```diff
@@ -45,16 +45,17 @@ func (ev *Event) GetPathResolutionError() error {
...
 	if len(f.PathnameStr) == 0 {
-		path, err := ev.resolvers.resolveInode(&f.FileFields)
+		path, err := ev.resolvers.resolveFileFieldsPath(&f.FileFields)
 		if err != nil {
 			if _, ok := err.(ErrTruncatedSegment); ok {
 				ev.SetPathResolutionError(err)
 			} else if _, ok := err.(ErrTruncatedParents); ok {
 				ev.SetPathResolutionError(err)
 			}
+			f.PathResolutionError = err
```
Comment (Gui774ume) [inline pkg/security/probe/model.go:58]: If we export resolution errors that are not `ErrTruncatedParents` or `ErrTruncatedSegment`, we might need to change the `NewAbnormalPathEvent` in `custom_events.go`

File: pkg/security/probe/tags_resolver.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	go func() {
+		<-ctx.Done()
+		t.tagger.Stop()
+	}()
+
+	return nil
+}
+
+// ResolverTags returns the tags for the given id
+func (t *TagsResolver) Resolve(id string) []string {
```
Comment (Gui774ume) [inline pkg/security/probe/tags_resolver.go:61]: We already discussed this on Slack, but we should probably remove the `ResolveLabels` method of the `ContainerResolver` :)

File: pkg/security/probe/process_resolver.go [hunk 4]
```diff
@@ -240,8 +241,8 @@ func (p *ProcessResolver) enrichEventFromProc(entry *model.ProcessCacheEntry, pr
 	if err != nil {
 		return errors.Wrapf(err, "snapshot failed for %d: couldn't parse kernel capabilities", proc.Pid)
 	}
-	_ = p.resolvers.ResolveProcessContextUser(&entry.ProcessContext)
-	_ = p.resolvers.ResolveProcessContextGroup(&entry.ProcessContext)
+	p.SetProcessUsersGroups(entry)
```
Comment (Gui774ume) [inline pkg/security/probe/process_resolver.go:244]: Couldn't add a comment on the right line with my IDE, but we should change `len(filledProc.Gids) > 0` to `len(filledProc.Gids) > 4`.

File: pkg/tagger/remote/tagger.go [hunk 5]
```diff
@@ -266,6 +266,8 @@ func (t *Tagger) processResponse(response *pb.StreamTagsResponse) error {
 				StandardTags:                entity.StandardTags,
 			},
 		})
+
+		fmt.Printf("QQQQQQQQQQQQqqq: %+v\n", convertEntityID(entity.Id))
```
Comment (Gui774ume) [inline pkg/tagger/remote/tagger.go:270]: Haha I think you forgot this one :)

File: .gitlab/source_test/ebpf.yml (hunk 1 above)
Comment (safchain) [reply .gitlab/source_test/ebpf.yml:38]: Indeed was just for debugging. Will removed it

**Review by Gui774ume — APPROVED** [review]
lgtm

**Review by lebauce — APPROVED** [review]

**Pull request merged**